[UI] Rework of overview page

Warrows opened the pull request:
Here is a proposal for some modifications on the overview page. Thanks to @Mrs-X work it got some really nice improvements but I believe it still holds a few flaws. So here are my proposed changes:

-First, put the combined balance on top. Users want to know how much they have, and how much they can spend right away. Stop.
-For the same reason, remove emphasis from piv balance and put it to combined. 
-Move PIVX/zPIV percentages to the areas labels. (piv one was in front of unlocked which was technically false)
-Change sections names for more consistency (plurals, zPIV instead of zerocoin)
-Move locked to the PIVX balance instead of combined (you can't lock zPIV, can you ?)
-Remove redundant informations like zPIV and PIVX balances in both their own areas and the combined balance area. 
-Reverse zPIV balances order to be consistent with the two other sections.
-Have a clear available/total policy for each section.
  
Here is a small before/after comparison. 
![before](https://user-images.githubusercontent.com/835098/34639813-770ffe34-f2e7-11e7-872b-7941a22b2755.png)
![after](https://user-images.githubusercontent.com/835098/34639815-7ad034e4-f2e7-11e7-8ac6-96ba6b65f3c8.png)

I believe all previously available information are still in here.

presstab commented:
I like the concept here. utACK. Need to confirm that this reflects all balances correctly.

Fuzzbawls commented on src/qt/forms/overviewpage.ui:
```diff
@@ -546,21 +661,87 @@
+                         </property>
                          <property name="text">
-                          <string>Spendable:</string>
+                          <string>0.000 000 00 PIV</string>
```
should be `<string notr="true">0.000 000 00 PIV</string>` to prevent sending this to transifex for translation

Warrows replied:
Same here.

Fuzzbawls commented on src/qt/forms/overviewpage.ui:
```diff
@@ -638,66 +839,38 @@
                          </property>
                          <property name="text">
-                          <string notr="true">0.000 000 00 PIV</string>
+                          <string>0.000 000 00 PIV</string>
```
should keep the `notr="true"` here

Warrows replied:
Same. It was not voluntarily removed. I just moved parts around. Will add.

Fuzzbawls commented on src/qt/overviewpage.cpp:
```diff
@@ -202,40 +202,47 @@ void OverviewPage::setBalance(const CAmount& balance, const CAmount& unconfirmed
+    ui->labelUnconfirmed->setText(BitcoinUnits::floorHtmlWithUnit(  nDisplayUnit, unconfirmedBalance,  false, BitcoinUnits::separatorAlways));
+    ui->labelImmature->setText(BitcoinUnits::floorHtmlWithUnit(     nDisplayUnit, immatureBalance,     false, BitcoinUnits::separatorAlways));
+    ui->labelTotal->setText(BitcoinUnits::floorHtmlWithUnit(        nDisplayUnit, nTotalBalance,       false, BitcoinUnits::separatorAlways));
+    ui->labelLockedBalance->setText(BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, nLockedBalance,      false, BitcoinUnits::separatorAlways));
```
NIT: unnecessary whitespace in function parameters

Warrows replied:
About this, I will remove them for the sake of file consistency. But there is a serious question. I personally find this area more readable with this whitespaces. Other areas would benefit from the same treatment in my view. They obviously have no influence on compiled code. 
But is there coding style guidelines and where can they be found if so?

Fuzzbawls commented on src/qt/overviewpage.cpp:
```diff
@@ -202,40 +202,47 @@ void OverviewPage::setBalance(const CAmount& balance, const CAmount& unconfirmed
+    ui->labelWatchAvailable->setText(BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, watchOnlyBalance,     false, BitcoinUnits::separatorAlways));
+    ui->labelWatchPending->setText(BitcoinUnits::floorHtmlWithUnit(  nDisplayUnit, watchUnconfBalance,   false, BitcoinUnits::separatorAlways));
+    ui->labelWatchImmature->setText(BitcoinUnits::floorHtmlWithUnit( nDisplayUnit, watchImmatureBalance, false, BitcoinUnits::separatorAlways));
+    ui->labelWatchTotal->setText(BitcoinUnits::floorHtmlWithUnit(    nDisplayUnit, nTotalWatchBalance,   false, BitcoinUnits::separatorAlways));
```
NIT: unnecessary whitespace in function parameters

Fuzzbawls commented on src/qt/overviewpage.cpp:
```diff
@@ -202,40 +202,47 @@ void OverviewPage::setBalance(const CAmount& balance, const CAmount& unconfirmed
-    ui->labelzBalancez->setText(BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, zerocoinBalance, false, BitcoinUnits::separatorAlways));
-    ui->labelzBalanceImmature->setText(BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, immatureZerocoinBalance, false, BitcoinUnits::separatorAlways));
+    ui->labelzBalance->setText(           BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, zerocoinBalance,            false, BitcoinUnits::separatorAlways));
+    ui->labelzBalanceImmature->setText(   BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, immatureZerocoinBalance,    false, BitcoinUnits::separatorAlways));
```
NIT: unnecessary whitespace in function parameters

Fuzzbawls commented on src/qt/overviewpage.cpp:
```diff
@@ -202,40 +202,47 @@ void OverviewPage::setBalance(const CAmount& balance, const CAmount& unconfirmed
+    CAmount sumTotalBalance         = nTotalBalance + zerocoinBalance;
+
+    ui->labelBalancez->setText(BitcoinUnits::floorHtmlWithUnit(nDisplayUnit, availableTotalBalance,   false, BitcoinUnits::separatorAlways));
+    ui->labelTotalz->setText(BitcoinUnits::floorHtmlWithUnit(  nDisplayUnit, sumTotalBalance,         false, BitcoinUnits::separatorAlways));
```
NIT: unnecessary whitespace in function parameters

Fuzzbawls commented on src/qt/forms/overviewpage.ui:
```diff
@@ -352,40 +362,130 @@
+                       <item>
+                        <widget class="QLabel" name="labelPIVPercent">
+                         <property name="text">
+                          <string>0 %</string>
```
should probably add `notr="true"` to the `<string>` element here.

Warrows replied:
I am pretty sure this was not everywhere before. Will add of course.

Mrs-X commented on src/qt/forms/overviewpage.ui:
```diff
@@ -239,11 +245,14 @@
+                         </property>
                          <property name="text">
-                          <string>PIV Balances</string>
+                          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Combined Balances&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
```
Yep, missed this one.
Please remove the HTML code and just increase the font size for this label.
a) translation is easier, and it's b) also easier to change the font properties in the CSS file if needed.

Warrows replied:
Yeah I don't know where this comes from. Probably a mistake with qt creator. Will remove.

Warrows commented on src/qt/forms/overviewpage.ui:
```diff
@@ -239,11 +245,14 @@
                          </property>
                         </widget>
-                       </item>
+                       </item>                       
```
Text editor fizzling, will remove.

Warrows commented on src/qt/forms/overviewpage.ui:
```diff
@@ -352,40 +362,130 @@
-                         <property name="cursor">
-                          <cursorShape>IBeamCursor</cursorShape>
+                         <property name="text">
+                          <string>PIVX Balances</string>
```
Will change back.

Warrows commented:
Thanks to thorough review by @Fuzzbawls I got to make some changes.
I added a few "notr" tags which where absent in the original file.
Fixed a few editor fizzles.
And I rolled back some coding style changes regarding whitespaces. But I am left wondering if and where we have coding style guidelines.

Mrs-X commented:
> where we have coding style guidelines.

Not complete, but gets you started:
https://github.com/PIVX-Project/PIVX/blob/master/doc/developer-notes.md

wholewheatbiscuits commented:
Much cleaner, thanks! In last image,

Suggest auto-hiding the following if they are 0: Pending, Immature, Unconfimed, Locked.

Suggest removing Total and Available if they equal the same number, and only show the number.

Correction: PIVX Balances should be PIV Balance, or suggest changing to PIV
Correction: zPIV Balances should  be zPIV Balance, or suggest changing to zPIV

Warrows commented:
So yeah 7a8ad53 Implements changes suggested by @wholewheatbiscuits 
It also hides PIV and zPIV areas when there are none.
I also made sure that watch-only addresses are working as pointed out by @Fuzzbawls 
Here are the latest screenshots:
When opening a new (empty) wallet
![image](https://user-images.githubusercontent.com/835098/34684650-1e1cae60-f4a6-11e7-9769-b4b63fb2864a.png)
When using it like a "normal" user
![image](https://user-images.githubusercontent.com/835098/34684722-53dfdfea-f4a6-11e7-8059-799140c08a38.png)
And finally when you like fancy stuff like masternodes and watch-only addresses
![image](https://user-images.githubusercontent.com/835098/34684930-08a80eac-f4a7-11e7-8c83-2af68b8c9b96.png)
Obviously no configuration needed whatsoever and everything you own is always visible. I am just hiding empty balances.

turnkit commented:
Warrows - great improvement.
Three concerns:
1) tpiv - is this for "test network pivs" or "thousand pivs" or "total pivs?" Either way, it's confusing.  (And hopefully mPIVs and uPIVs eventually can get clever arbitrary names like milliPIVs, microPIVs,  'Doriens', 'Warrows', 'PIVpennies' or 'PIVbits', etc. so these units show up in more human friendly names.)
2) I'd suggest the "simple view" just show two numbers:  avail. PIVs & avail. zPIVs.
3) when there is not "detailed view" to be seen, still allow the user a way to twiddle/click an arrowhead to see all the "0.00" entries and their descriptions -- but that by default when they are all zero, that arrow is up/closed / view is not seen.

Warrows commented:
@turnkit Thanks for your valuable input.
tPIV stands for testnetPIV, no worries. About units, it's an interesting question but not to be solved here. I believe it's a community wide question which will need further discussions.
It might be unclear but there is no "simple view" here. It's one and only one panel which shows only numbers above 0. So if you have no pending operation and no locked up PIV in any way you get what's seen in the second image: Total (which happen to also be available), PIV and zPIV.
I am not sure there is a point in showing all the empty balances. If you are a new user you most likely don't care about them and if you are more a advanced one you probably already know about them and don't want it to cluster your view for nothing. 
Anyway you'll see them soon enough if you move funds around. And if they don't show long it's that the network is so fast you don't need to know about them.

turnkit commented:
Thanks for the positive comments.

I would disagree about need to keep the zero balances out of view. By
allowing a simple view vs. complete view the users who wish to learn by
observation have that opportunity.  This especially becomes useful when
roll-over help text appears describing what different balances mean.  So by
allowing a user to "open up" the complete detailed view they are then able
to learn before it gets complicated, when there is no frustration over,
"why can't I spend all the coin I have?!" when coins that were available
stop becoming available, then become available again, only leaving a user
to wonder, "what is happening?"  In those situations it will be useful for
a user to open up the detailed view and look at the metrics that were
populated when a coin became immature due to minting, right?

If that's not clear, I could probably re-write it so that it make more
sense.  The basic principle here is that the user should be King/Queen:
they should have the ability to get into all the details even when it's not
primarily relevant, if they wish to.  And primarily they should be able to
do so using methods that increase their knowledge and understanding by
doing so.



On Mon, Jan 8, 2018 at 2:09 PM, Warrows <notifications@github.com> wrote:

> @turnkit <https://github.com/turnkit> Thanks for your valuable input.
> tPIV stands for testnetPIV, no worries. About units, it's an interesting
> question but not to be solved here. I believe it's a community wide
> question which will need further discussions.
> It might be unclear but there is no "simple view" here. It's one and only
> one panel which shows only numbers above 0. So if you have no pending
> operation and no locked up PIV in any way you get what's seen in the second
> image: Total (which happen to also be available), PIV and zPIV.
> I am not sure there is a point in showing all the empty balances. If you
> are a new user you most likely don't care about them and if you are more a
> advanced one you probably already know about them and don't want it to
> cluster your view for nothing.
> Anyway you'll see them soon enough if you move funds around. And if they
> don't show long it's that the network is so fast you don't need to know
> about them.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PIVX-Project/PIVX/pull/479#issuecomment-356081455>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD711yxjfEoQYGlVILqOGT7urMCRiXrRks5tInYCgaJpZM4RVTEW>
> .
>



-- 
David C. Sutherland   |  (310) 729-6411 (c)  |  Lindale, Texas

Warrows commented:
@turnkit I absolutely understand your feeling that the user must be king and I agree with it. The thing is I don't think the average user will open the advanced view if he doesn't need it. And even if he did, there is no way we could stuff enough information in the software. If you want to understand cryptocurrencies you need to go out and educate yourself about them. As much as we would want it, we can't give a full course about blockchain and zPIV minting in-wallet. 
And in the case you just want to use the wallet without bothering too much about its internal workings, the fact that the locked coins show up when they are locked, with the much needed tooltip explaining why/how they are locked seems good enough to me.
You say yourself, it becomes useful to read about them when this situations happen (and they will happen, and the balances will be there, no click needed). Before their coins are locked, most users don't want to know what can happen to them. And those who want will learn on our website or youtube.

Warrows commented:
@presstab Thanks, totally agree. I just followed the existing pattern here, now it's done for all the same looking functions.

Mrs-X commented:
I'm not 100% sure if completely hiding elements when they are empty is support-wise a good idea.
We can't use that space anyway...

Have you ever tried to guide someone via phone through a problem who sees a different UI?

Warrows commented:
@Mrs-X I can't say I am 100% sure it's the best idea. But I really like the cleanliness it brings. 
I am not sure either about the different UI thing, it's not like hiding a button or putting things elsewhere depending on OS. It's more about putting away empty information. If you try to help someone, you know it's either there or 0.
Maybe let's try to get some support people input on this.
If there is a consensus I can rollback the hiding thing. But I think it does not make much sense to have it for some lines (was existing for pending, and maybe immature in PIV balance) and not others.

turnkit commented:
I agree.

I prefer the idea of two panels or a dashboard with a line between top,
minimal info, and bottom, details.

My idea is that when the details are zero the section Auto-rolls up, or
perhaps just has a green light for “all confirmed/processed/available”
whatever.

But when coins are unavailable somehow those numbers are highlighted.

But still the essential two numbers, available PIV and available zPIV
should be set apart and prominent, perhaps in much larger font.

On Tue, Jan 9, 2018 at 1:02 PM Mrs-X <notifications@github.com> wrote:

> I'm not 100% sure if completely hiding elements when they are empty is
> support-wise a good idea.
> We can't use that space anyway...
>
> Have you ever tried to guide someone via phone through a problem who sees
> a different UI?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/PIVX-Project/PIVX/pull/479#issuecomment-356381341>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD711w6bKtgLzODZ_6IQLZfj-oCvt4wlks5tI7e-gaJpZM4RVTEW>
> .
>
-- 
Sent from Gmail Mobile

turnkit commented:
It was very disturbing the first time I saw my balance go down, due to
minting and immature coins that were previously mature.

So any way the UI can calm the user when the previously available coins
disappear to a temporarily unavailable status, it is much better.

On Tue, Jan 9, 2018 at 4:36 PM David Sutherland <turnkit@gmail.com> wrote:

> I agree.
>
> I prefer the idea of two panels or a dashboard with a line between top,
> minimal info, and bottom, details.
>
> My idea is that when the details are zero the section Auto-rolls up, or
> perhaps just has a green light for “all confirmed/processed/available”
> whatever.
>
> But when coins are unavailable somehow those numbers are highlighted.
>
> But still the essential two numbers, available PIV and available zPIV
> should be set apart and prominent, perhaps in much larger font.
>
> On Tue, Jan 9, 2018 at 1:02 PM Mrs-X <notifications@github.com> wrote:
>
>> I'm not 100% sure if completely hiding elements when they are empty is
>> support-wise a good idea.
>> We can't use that space anyway...
>>
>> Have you ever tried to guide someone via phone through a problem who sees
>> a different UI?
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/PIVX-Project/PIVX/pull/479#issuecomment-356381341>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AD711w6bKtgLzODZ_6IQLZfj-oCvt4wlks5tI7e-gaJpZM4RVTEW>
>> .
>>
> --
> Sent from Gmail Mobile
>
-- 
Sent from Gmail Mobile

Mrs-X commented:
> If there is a consensus I can rollback the hiding thing.

@Warrows What about a new setting "Hide zero balances" to make this optional?
Could be used at other places as well.

Warrows commented:
@Mrs-X We can add a setting of course. But the point being to have a cleaner overview for newcomers the auto-hiding should be opt-out. Would you agree with that ?
Because in my view it's not a cosmetic change for power users but an effort to make the wallet more accessible.

Mrs-X commented:
@Warrows I just played around with different options.
I think your first screenshot (only Combined Balance) goes a bit too far, in the user, even when he's a beginner, should see that something like PIV and zPIV exist, even when one or both them are zero.

So if a more minimalistic overview should be the default I'd like see the second screenshot (this one:
https://user-images.githubusercontent.com/835098/34684650-1e1cae60-f4a6-11e7-9769-b4b63fb2864a.png).

wholewheatbiscuits commented:
@Mrs-X that's a good point, I didn't think about that. There's going to be edge cases where people think PIV is automatically private, but if they see zPIV next to it on the overview tab, that should make many understand they need zPIV for private transactions.
On 1/13/2018 11:45:56 AM, Mrs-X <notifications@github.com> wrote:
@Warrows [https://github.com/warrows] I just played around with different options.
I think your first screenshot (only Combined Balance) goes a bit too far, in the user, even when he's a beginner, should see that something like PIV and zPIV exist, even when one or both them are zero.
So if a more minimalistic overview should be the default I'd like see the second screenshot (this one:
https://user-images.githubusercontent.com/835098/34684650-1e1cae60-f4a6-11e7-9769-b4b63fb2864a.png [https://user-images.githubusercontent.com/835098/34684650-1e1cae60-f4a6-11e7-9769-b4b63fb2864a.png]).
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub [https://github.com/PIVX-Project/PIVX/pull/479#issuecomment-357460883], or mute the thread [https://github.com/notifications/unsubscribe-auth/Aczfkl6Vn0qE89Ks2Dnr09PYsf5pNmF-ks5tKQfygaJpZM4RVTEW].

Warrows commented:
@Mrs-X You are absolutely right, newcomers must see that there are two separate balances. Maybe hiding everything was a bit extreme.
I implemented the setting as opt-out in f751e12
And put back the two main panels as always visible in 778d9f7

Warrows commented:
I think I know exactly what the problem is, I am going to push that tomorrow. Do you want me to rebase and squash some commits by the way ?

Warrows commented:
@Mrs-X I made the resize work like before. I personally prefer the "fixed size" version. But that's not the point of this PR.

Mrs-X commented:
> I personally prefer the "fixed size" version.

Lots of people specifically asked to have the resizing like this, and it makes sense if we want to make use of bigger fonts to support 4k or 8k displays somewhere in the future.

I'll test later, we should do squashing when everything is ACKed.

Mrs-X commented:
ACK 👍 

When it's squashed and @Fuzzbawls approves his requested changes we'll merge.

Mrs-X commented:
ACK https://github.com/PIVX-Project/PIVX/pull/479/commits/b5f420bb1bc2dbcbddece72632183a13bae870c9

Fuzzbawls commented:
ACK b5f420b

Sieres commented:
ACK for the hide zero balances. Balance sections do no collapse like @Warrows examples
